rotkehlchen/data/importer.py

```diff
@@ -1014,3 +1015,167 @@ def import_shapeshift_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
+        """
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr='%a %b %d %Y %H:%M:%S %Z%z',
+            location='uphold',
+        )
+        destination = csv_row['Destination']
+        destination_asset = symbol_to_asset_or_token(csv_row['Destination Currency'])
+        destination_amount = deserialize_asset_amount(csv_row['Destination Amount'])
+        origin = csv_row['Origin']
+        origin_asset = symbol_to_asset_or_token(csv_row['Origin Currency'])
+        origin_amount = deserialize_asset_amount(csv_row['Origin Amount'])
+        fee = deserialize_fee(csv_row['Fee Amount']) or FVal('0.0')
+        fee_asset = symbol_to_asset_or_token(csv_row['Fee Currency'] or csv_row['Origin Currency'])
+        transaction_type = csv_row['Type']
+        notes = f"""
+Activity from uphold with uphold transaction id:
+ {csv_row['Id']}, origin: {csv_row['Origin']},
+ and destination: {csv_row['Destination']}.
+  Type: {csv_row['Type']}.
+  Status: {csv_row['Status']}.
+"""
+        # On Exchange Transactions and Trades.
+        if origin == destination == 'uphold':
+            # Transfers
+            if origin_asset == destination_asset:
+                # Transfers have the same amounts
+                if origin_amount == destination_amount:
+                    if transaction_type == 'in':
+                        action_type = LedgerActionType.INCOME
+                    elif transaction_type == 'out':
+                        if destination_asset == A_BAT:
+                            action_type = LedgerActionType.GIFT
+                        else:
+                            action_type = LedgerActionType.EXPENSE
+                    else:
+                        log.debug(f'Ignoring unmapped uphold internal transaction. {csv_row}')
+                        return
+                    action = LedgerAction(
+                        identifier=0,
+                        timestamp=timestamp,
+                        action_type=action_type,
+                        location=Location.UPHOLD,
+                        amount=destination_amount,
+                        asset=destination_asset,
+                        rate=None,
+                        rate_asset=None,
+                        link="",
+                        notes=notes,
+                    )
+                    self.db_ledger.add_ledger_action(action)
+            # Trades (buy)
+            else:
+                if transaction_type == 'transfer':
+                    trade = Trade(
+                        timestamp=timestamp,
+                        location=Location.UPHOLD,
+                        base_asset=destination_asset,
+                        quote_asset=origin_asset,
+                        trade_type=TradeType.BUY,
+                        amount=destination_amount,
+                        rate=Price(origin_amount / destination_amount),
```

Comment:
If `destination_amount` is ZERO here then you will get a division by zero error and the whole task will die. Please make sure it's not zero in this case and if it is skip accordingly perhaps with a log error.